Documentation/Coherency.md

```diff
@@ -0,0 +1,74 @@
+# Problem #
+-----------
+Official non-servicing builds of the .NET Core SDK are *coherent*, meaning that for each repo that contributes bits to the final product, exactly one version of that repo is used at all stages of the build.  For instance, CLI and core-setup both depend on CoreFX.  In a coherent build, this will be the same version of CoreFX, but in an incoherent build, CLI could have been built with a different version than core-setup.  This can cause problems which usually manifest in source-build as build errors along the lines of `Could not load file or assembly 'NuGet.Frameworks, Version=4.9.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The located assembly's manifest definition does not match the assembly reference.`, but could also result in runtime errors.  
+Source-build has traditionally worked around this by forcing all repos to use the same bits (the source-built ones) throughout the build process.  However, this brings in new problems:
+- We are not necessarily shipping the exact bits that an official release would.
+- Servicing releases *should* be incoherent.  We do not ship non-serviced packages, so a valid official SDK could have some 3.0.1 bits and some 3.0.0 bits.
+	- This also means that we do not get all the packages we need from our forced-coherent build.
+- Very few official builds are coherent during development.  Builds diverge from coherence within days after release and soon become massively incoherent (dependency graph reports 44 distinct repo-shas for today's CoreFX build, and 181 distinct repo-shas for core-sdk).  We would like to support daily builds, which means that we have to deal with incoherence during the development process.
```

Comment:
> dependency graph reports 44 distinct repo-shas for today's CoreFX build,

Can you clearify the "for"? I'm assuming it means CoreFX depends on 44 repo+commit combinations as upstreams, but it's easy to read it as 44 commits of CoreFX being needed to build downstreams, or maybe something else.

Reply:
Clarified the language a bit.